### PR TITLE
VideoPlayer: rework VideoBuffers

### PR DESCRIPTION
**Pull request by FernetMenta**

### CODE REVIEW

File: xbmc/cores/VideoPlayer/Process/VideoBuffer.h [hunk 1]
```diff
@@ -102,7 +98,7 @@ class CVideoBuffer
   virtual AVPixelFormat GetFormat();
   virtual void GetPlanes(uint8_t*(&planes)[YuvImage::MAX_PLANES]) {};
   virtual void GetStrides(int(&strides)[YuvImage::MAX_PLANES]) {};
-  virtual void SetDimensions(int width, int height, int alignedWidth, int alignedHeight) {};
+  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES]) {};
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: @popcornmix you need an adjustment here

File: xbmc/cores/VideoPlayer/Process/VideoBuffer.cpp [hunk 2]
```diff
@@ -263,9 +266,14 @@ bool CVideoBufferSysMem::Alloc()
     m_image.planesize[2] = 0;
   }
 
-  for (int i = 0; i < 3; ++i)
-    m_image.plane[i] = new uint8_t[m_image.planesize[i]];
+  m_image.plane[0] = m_data;
+  m_image.plane[1] = m_data + m_image.planesize[0];
+  m_image.plane[2] = m_data + m_image.planesize[1];
```
Comment (popcornmix) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.cpp]: This doesn't look right. Perhaps:
m_image.plane[2] = m_image.plane[1] + m_image.planesize[1];

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp [hunk 3]
```diff
@@ -119,12 +119,15 @@ bool CDVDVideoPPFFmpeg::Process(VideoPicture* pPicture)
...
   }
 
-  m_pTarget.videoBuffer = m_processInfo.GetVideoBufferManager().Get(AV_PIX_FMT_YUV420P, pPicture->iWidth * pPicture->iHeight * 3 / 2);
+  m_pTarget.videoBuffer = m_processInfo.GetVideoBufferManager().Get(AV_PIX_FMT_YUV420P, pPicture->iWidth * pPicture->iHeight);
   if (!m_pTarget.videoBuffer)
   {
     return false;
   }
 
+  int strides[YuvImage::MAX_PLANES];
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp]: Use `int strides[YuvImage::MAX_PLANES] = { };` to zero out the unused strides

File: xbmc/cores/VideoPlayer/Process/VideoBuffer.cpp (hunk 2 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.cpp]: right, thanks

File: xbmc/cores/VideoPlayer/Process/VideoBuffer.h [hunk 4]
```diff
@@ -119,15 +115,19 @@ class CVideoBuffer
...
   virtual ~CVideoBufferSysMem();
   virtual void GetPlanes(uint8_t*(&planes)[YuvImage::MAX_PLANES]) override;
   virtual void GetStrides(int(&strides)[YuvImage::MAX_PLANES]) override;
+  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES]) override;
   bool Alloc();
 
 protected:
   int m_width = 0;
   int m_height = 0;
+  int m_strides[YuvImage::MAX_PLANES];
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: Is it possible to do `int m_strides[YuvImage::MAX_PLANES] = { };`?

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp (hunk 3 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp]: ok

File: xbmc/cores/VideoPlayer/Process/VideoBuffer.h (hunk 4 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: I will try something to init here.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: seems to work

File: xbmc/cores/VideoPlayer/Process/VideoBuffer.h [hunk 5]
```diff
@@ -70,10 +70,6 @@ class IVideoBufferPool : public std::enable_shared_from_this<IVideoBufferPool>
   // as soon as it knows parameters: pixFmx, size
   virtual void Configure(AVPixelFormat format, int size) {};
 
-  // configure dimensions of pool. May be called after first buffer is allocated
-  // so renderer knows dimensions of buffer
-  virtual void SetDimensions(int width, int height, int alignedWidth, int alignedHeight) {};
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: @popcornmix calling this on the pool makes not too much sense imo. a pool gets configured with a size and as long a pic fits into size the pool does not care anymore for width, height, or strides

**Pull request merged**